### DISCUSSION TITLE
Question about the beginner tutorial

### DISCUSSION THREAD

**Discussion by SnuK87** [deepjavalibrary/djl#3547] [Q&A]
Hi,

today I was going through the beginner tutorial and faced some difficulties in the final image classification process.
I followed [this document](https://github.com/deepjavalibrary/djl/blob/master/examples/docs/train_mnist_mlp.md) for the training and afterwards [this document](https://github.com/deepjavalibrary/djl/blob/master/examples/docs/image_classification.md) for the image classification.

When I use the provided `0.png` file I get the same result as shown in the document but when I use an other images the recognition is not working.
 
For example: When I run this image with the digit `6` I get the following result:
![6_test](https://github.com/user-attachments/assets/b430cc52-4747-4f8a-9d1c-49c5fc3da8df)

```
[
	{"className": "5", "probability": 0,83956},
	{"className": "6", "probability": 0,15970},
        ...
]
```

Here is the log output of the model I used for the classification:
```
Model (
	Name: mlp
	Model location: .\build\model
	Data Type: float32
	Accuracy: 0,97440
	Loss: 0,08300
	Epoch: 5
)
```

I only get a good result when I use .png files that are part of the dataset. Is this normal or am I missing something here?

**Comment by frankfliu** [comment]
I think this is normal. first of all "5", "6" is similar and result isn't far off.

The mlp model is pretty basic, and the hidden layers in the example is pretty small.
